### PR TITLE
TC-02: Setup datastore with redux

### PR DESCRIPTION
**Pull request by Haris-moin**
- setup redux store
- create redux structure
- implement redux actions , reducer and state
- test redux actions and state in example component

### CODE REVIEW

File: src/store/modules/example/reducer.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Action } from "./action";
+import { EXAMPLE_REQUEST_TYPES } from "./types";
+
+export const exampleReducer = (state = 0, action: Action) => {
+  switch (action.type) {
+    case EXAMPLE_REQUEST_TYPES.EXAMPLE_INCREAMENT_COUNTER:
+      return state + 1;
```
Comment (hiqbalmallick) [inline src/store/modules/example/reducer.ts]: This is not the right way to update the state in the reducer.

File: src/example/counter/index.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+const Example = () => {
+  const dispatch = useDispatch();
+  const count = useSelector(({ exampleReducer }: RootState) => exampleReducer);
+
+  return (
+    <div>
+      <button
+        onClick={() => {
+          dispatch(increamentCouter());
+        }}
```
Comment (hiqbalmallick) [inline src/example/counter/index.tsx]: create a method above and pass the reference here

File: src/example/counter/index.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      >
+        Add
+      </button>
+      <h2>Count: {count}</h2>
+      <button
+        onClick={() => {
+          dispatch(decreamentCouter());
+        }}
```
Comment (hiqbalmallick) [inline src/example/counter/index.tsx]: pass method reference here

File: src/store/modules/example/reducer.ts (hunk 1 above)
Comment (Haris-moin) [reply src/store/modules/example/reducer.ts]: resolved it in correct way

File: src/example/counter/index.tsx (hunk 2 above)
Comment (Haris-moin) [reply src/example/counter/index.tsx]: resolved it by passing a reference

File: src/example/counter/index.tsx (hunk 3 above)
Comment (Haris-moin) [reply src/example/counter/index.tsx]: resolved it by passing a reference

**Review by hiqbalmallick — APPROVED** [review]

**Pull request merged**